openpype/pipeline/colorspace.py

```diff
@@ -761,21 +763,32 @@ def get_imageio_config(
     Args:
         project_name (str): project name
         host_name (str): host name
+        asset_name (Optional[str]): asset name. Defaults to None.
+        task_name (Optional[str]): task name. Defaults to None.
         project_settings (Optional[dict]): Project settings.
-        anatomy_data (Optional[dict]): anatomy formatting data.
         anatomy (Optional[Anatomy]): Anatomy object.
         env (Optional[dict]): Environment variables.
+        anatomy_data (Optional[dict]): anatomy formatting data.
 
     Returns:
         dict: config path data or empty dict
     """
+    from openpype.pipeline import get_current_context
+    from openpype.pipeline.template_data import get_template_data_with_names
+
+    context = get_current_context()
+
     project_settings = project_settings or get_project_settings(project_name)
     anatomy = anatomy or Anatomy(project_name)
 
+    if project_name == context["project_name"] and not asset_name:
+        asset_name = context["asset_name"]
+        if not task_name:
+            task_name = context["task_name"]
```

Comment:
Only need to get the current context if asset name is not set.
```suggestion
    project_settings = project_settings or get_project_settings(project_name)
    anatomy = anatomy or Anatomy(project_name)
    
    if asset_name is None:
        context = get_current_context()
        if project_name == context["project_name"]:
            asset_name = context["asset_name"]
            if task_name is None:
                task_name = context["task_name"]
```

Reply:
hey @BigRoy thanks for suggestions. We were discussing it with @iLLiCiTiT and there was recommended to make the context related inputs required for the future proofed workflow. I agree that we should be able to call some of those functions sometimes, for example, in Editorial where current context is set to edit task but we will be publishing into different shot context. There we will rather benefit with explicit context inputs.

Reply:
I am resolving the  new commit is going around this suggestion.